Clients/Xamarin.Interactive.Client.Mac/ViewInspector/InspectViewNode.cs

```diff
@@ -7,130 +7,47 @@
 
 using System;
 using System.Collections.Generic;
-
+using System.ComponentModel;
+using System.Linq;
 using AppKit;
 using CoreGraphics;
 using Foundation;
 using SceneKit;
 
+using Xamarin.Interactive.Client.ViewInspector;
 using Xamarin.Interactive.Inspection;
 using Xamarin.Interactive.Remote;
 
 namespace Xamarin.Interactive.Client.Mac.ViewInspector
 {
-    class TreeState
-    {
-        public int TotalCount { get; private set; }
-        public DisplayMode Mode { get; }
-        public bool ShowHidden { get; }
-
-        List<int> ancestors = new List<int> { 0 };
-
-        public TreeState (DisplayMode mode, bool showHidden)
-        {
-            Mode = mode;
-            ShowHidden = showHidden;
-        }
-
-        public void PushGeneration ()
-        {
-            ancestors.Add (0);
-        }
-
-        public int AddChild (InspectView view)
-        {
-            var pos = ancestors.Count - 1;
-            var count = ancestors [pos];
-            ancestors [pos] = count + 1;
-            TotalCount = TotalCount + 1;
-
-            return count;
-        }
-
-        public int PopGeneration ()
-        {
-            var pos = ancestors.Count - 1;
-            var count = ancestors [pos];
-
-            ancestors.RemoveAt (ancestors.Count - 1);
-            return count;
-        }
-    }
-
-    sealed class InspectViewNode : SCNNode, IInspectViewNode
+    sealed class InspectViewNode : SCNNode, IInspectTree3DNode<InspectViewNode>
     {
         const float ZSpacing = 20f;
         const float ZFightDenominator = 20f;
 
-        public InspectView InspectView { get; }
+        public InspectTreeNode Node { get; private set; }
```

Comment:
Does this get set outside the constructor? I don't see it...

Reply:
It gets set in `public InspectViewNode`, just below--`Node = node`.

Reply:
Right, in the constructor. So `private set` isn't needed?